groups/instance/locals.tf

```diff
@@ -12,45 +12,33 @@ locals {
     values(data.aws_subnet.placement).*.availability_zone,
     values(data.aws_subnet.placement).*.id
   ))
-  placement_subnet_pattern = local.secrets.placement_subnet_pattern
-  placement_vpc_pattern    = local.secrets.placement_vpc_pattern
-
   automation_subnet_ids = values(zipmap(
     values(data.aws_subnet.automation).*.availability_zone,
     values(data.aws_subnet.automation).*.id
   ))
+
+  placement_subnet_pattern  = local.secrets.placement_subnet_pattern
+  placement_vpc_pattern     = local.secrets.placement_vpc_pattern
   automation_subnet_pattern = local.secrets.automation_subnet_pattern
   automation_vpc_pattern    = local.secrets.automation_vpc_pattern
   concourse_access_cidrs    = local.secrets.concourse_access_cidrs
-
-  artifactory_web_access = concat(local.placement_subnet_cidrs, [local.concourse_access_cidrs])
-
-  dns_zone_name       = local.secrets.dns_zone_name
-  dns_zone_is_private = local.secrets.dns_zone_is_private
-
-  aws_route53_record_name = "${var.service}-${var.environment}.${data.aws_route53_zone.selected.name}"
-
-  ssl_certificate_name = local.secrets.ssl_certificate_name
-
-  create_ssl_certificate = local.ssl_certificate_name == "" ? true : false
-  ssl_certificate_arn    = local.ssl_certificate_name == "" ? aws_acm_certificate_validation.certificate[0].certificate_arn : data.aws_acm_certificate.certificate[0].arn
-
-  db_username = local.secrets.db_username
-  db_password = local.secrets.db_password
-  db_fqdn     = "${var.service}db.${data.aws_route53_zone.selected.name}"
-
-  ssh_keyname    = "${var.service}-${var.environment}"
-  ssh_public_key = local.secrets.public_key
-
-  ami_owner_id             = local.secrets.ami_owner_id
-  account_id               = local.secrets.ami_owner_id
-  artifactory_access_token = local.secrets.artifactory_access_token
+  artifactory_web_access    = concat(local.placement_subnet_cidrs, [local.concourse_access_cidrs])
+  dns_zone_name             = local.secrets.dns_zone_name
+  dns_zone_is_private       = local.secrets.dns_zone_is_private
+  aws_route53_record_name   = "${var.service}-${var.environment}.${data.aws_route53_zone.selected.name}"
+  ssl_certificate_name      = local.secrets.ssl_certificate_name
+  create_ssl_certificate    = local.ssl_certificate_name == "" ? true : false
+  ssl_certificate_arn       = local.ssl_certificate_name == "" ? aws_acm_certificate_validation.certificate[0].certificate_arn : data.aws_acm_certificate.certificate[0].arn
+  db_fqdn                   = "${var.service}-db.${data.aws_route53_zone.selected.name}:${local.secrets.db_port}"
+  ssh_keyname               = "${var.service}-${var.environment}"
+  ssh_public_key            = local.secrets.public_key
+  ami_owner_id              = local.secrets.ami_owner_id
+  account_id                = local.secrets.ami_owner_id
+  efs_access_point_id       = module.efs_file_system.efs_access_point_ids["${var.efs_artifacts_access_point_name}"].id
```

Comment:
I believe you could simplify this slightly to: `module.efs_file_system.efs_access_point_ids[var.efs_artifacts_access_point_name].id`